dynamic_rest/serializers.py

```diff
@@ -504,11 +568,44 @@ def _faster_to_representation(self, instance):
         ret = {}
         fields = self._readable_fields
 
+        is_fast = isinstance(instance, prefetch.FastObject)
+        id_fields = self._readable_id_fields
+
         for field in fields:
-            try:
-                attribute = field.get_attribute(instance)
-            except SkipField:
-                continue
+            # we exclude dynamic fields here because the proper fastquery
+            # dereferencing happens in the `get_attribute` method now
+            if (
+                is_fast and
+                not isinstance(
+                    field,
+                    (DynamicGenericRelationField, DynamicRelationField)
+                )
+            ):
+                if field in id_fields and field.source not in instance:
+                    # TODO - make better.
+                    attribute = instance.get(field.source + '_id')
+                    ret[field.field_name] = attribute
+                    continue
+                else:
+                    try:
+                        attribute = instance[field.source]
+                    except KeyError:
+                        # slower, but does more stuff
+                        # Also, some temp debugging
+                        if hasattr(instance, field.source):
```

Comment:
I think this check was for debugging. We could probably remove and just do `getattr(instance, field.source, None)`.

Reply:
I'm going to leave this in for now.  While making some modifications to the FastQuery code, I definitely saw these warning messages from time to time... They were helpful in debugging, since they usually arose when something was actually wrong.